File: employee-system/src/main/java/com/employeesystem/services/EmployeeService.java
package com.employeesystem.services;

import com.employeesystem.entity.Employee;
import com.employeesystem.model.EmployeeModel;

import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);

    List<EmployeeModel> getEmployees();

    boolean deleteEmployee(Long id);

    EmployeeModel getEmployeeById(Long id);

    EmployeeModel editEmployee(Long id, EmployeeModel employeeModel);
}

File: react-employee-system/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://18.192.51.52:8080/api/v1/employees";

class EmployeeService {
  saveEmployee(employee) {
    return axios.post(EMPLOYEE_API_BASE_URL, employee);
  }

  getEmployees() {
    return axios.get(EMPLOYEE_API_BASE_URL);
  }

  deleteEmployee(id) {
    return axios.delete(EMPLOYEE_API_BASE_URL + "/" + id);
  }

  getEmployeeById(id) {
    return axios.get(EMPLOYEE_API_BASE_URL + "/" + id);
  }

  editEmployee(employee, id) {
    return axios.put(EMPLOYEE_API_BASE_URL + "/" + id, employee);
  }
}

export default new EmployeeService();
